lib/genesis/helpers.go

```diff
@@ -225,25 +215,61 @@ func buildRawMap(m map[string]map[string]interface{}) (map[string]string, error)
 	return res, nil
 }
 
-func buildRawMapInterface(m map[string]interface{}, kv *keyValue) error {
-	for k, v := range m {
+func buildRawMapInterface(m interface{}, kv *keyValue) error {
+	mRefObjVal := reflect.Indirect(reflect.ValueOf(m))
+
+	for i := 0; i < mRefObjVal.NumField(); i++ {
+		jsonTag := mRefObjVal.Type().Field(i).Tag.Get("json")
+		k := strings.Split(jsonTag, ",")[0]
 		kv.key = append(kv.key, k)
-		switch v2 := v.(type) {
-		case []interface{}:
-			kv.valueLen = big.NewInt(int64(len(v2)))
-			if err := buildRawArrayInterface(v2, kv); err != nil {
-				return err
-			}
+		v := mRefObjVal.Field(i)
+
+		if v.Kind() == reflect.Struct && v.IsNil() {
+			continue
+		}
+
+		switch v2 := v.Interface().(type) {
 		case string:
 			kv.value = v2
+		case uint64, int64, int:
+			kv.value = fmt.Sprint(v2)
+		default:
+
+			switch v.Kind() {
+			case reflect.Slice:
+
+				vLen := v.Len()
+				listOfInterface := []interface{}{}
+				for i := 0; i < vLen; i++ {
+					listOfInterface = append(listOfInterface, v.Index(i).Interface())
+				}
+
+				if vLen > 0 && v.Index(0).Kind() == reflect.Struct {
+					kv.valueLen = big.NewInt(int64(v.Index(0).NumField()))
+				} else {
+					kv.valueLen = big.NewInt(int64(vLen))
+
+				}
+				if err := buildRawArrayInterface(listOfInterface, kv); err != nil {
+					return err
```

Comment:
```suggestion
					return fmt.Errorf("error building raw array interface: %w", err)
```

Since buildRawArrayInterface is a decent size function, it might be helpful to wrap the error to give extra info.

Also see other place where buildRawArrayInterface is getting using and we can wrap errors.